pkg/kv/kvserver/gc/gc_test.go

```diff
@@ -379,3 +397,561 @@ func TestGCIntentBatcherErrorHandling(t *testing.T) {
 	cancel()
 	require.Error(t, batcher.maybeFlushPendingIntents(ctx))
 }
+
+/*
+Table data format:
+(see data below for examples referenced in brackets after each explanation)
+
+Top row contains keys separated by one or more spaces [a to j].
+
+Data rows start with timestamp followed by values aligned with the first
+character of a key in top row.
+
+Timestamp could be prefixed with '>' to set GC timestamp [gc threshold is 5].
+Intersection of key and timestamp defines value for the key [a@9 == 'A',
+a@2 == 'c'].
+If value contains any upper case characters it is expected that its key should
+not be garbage collected [A, B, D, E].
+
+Special value of . means a tombstone that should be collected [a@4 should be
+garbage collected].
+Special value of * means a tombstone that should not be collected [tombstone c@8
+should not be garbage collected].
```

Comment:
We should have tests for tombstones with a non-empty `MVCCValueHeader` as well (i.e. with a `LocalTimestamp`), either here or elsewhere, to ensure GC still treats them as tombstones even though they have a non-empty value. We do have some metamorphic testing for this as well (see `mvcc-value-disable-simple-encoding`), but we don't hit that here because we're writing raw data straight to the engine rather than going via the high-level MVCC functions.

Reply:
Randomized tests hit that failure with metamorphic timestamp and failed!
I'll see if we can use high level functions here as well to have extra coverage.